gravitee-apim-plugin/gravitee-apim-plugin-endpoint/gravitee-apim-plugin-endpoint-mqtt5/src/main/java/io/gravitee/plugin/endpoint/mqtt5/Mqtt5EndpointConnector.java

```diff
@@ -249,7 +273,48 @@ private String getResponseTopic(final ExecutionContext ctx) {
         return responseTopic;
     }
 
+    private String overrideConfigFromMessage(final String sharedConfiguration, final Message message, final String attributeKey) {
+        String returnedConfiguration = sharedConfiguration;
+        String attributeConfiguration = message.attribute(attributeKey);
+        if (attributeConfiguration != null && !attributeConfiguration.isEmpty()) {
+            returnedConfiguration = attributeConfiguration;
+        }
+        if (returnedConfiguration != null) {
+            message.attribute(attributeKey, returnedConfiguration);
+        }
+        return returnedConfiguration;
+    }
+
+    private Flowable<Message> interruptMessagesWith(final ExecutionContext ctx, final Throwable throwable) {
+        if (
+            throwable instanceof ConnectionFailedException ||
+            throwable instanceof Mqtt5ConnAckException ||
+            throwable instanceof Mqtt5AuthException
+        ) {
+            return ctx.interruptMessagesWith(
+                new ExecutionFailure(500)
+                    .message("Endpoint connection failed")
```

Comment:
if we put some constants at interface level maybe this message could have its place there too ?

Reply:
For me, interface are not the good place to store constant. This text is only used here, don't think it is required a constant.